pkg/cmd/roachtest/tests/versionupgrade.go

```diff
@@ -417,7 +417,17 @@ func upgradeNodes(
 			newVersionMsg = "<current>"
 		}
 		t.L().Printf("restarting node %d into version %s", node, newVersionMsg)
-		c.Stop(ctx, t.L(), option.DefaultStopOpts(), c.Node(node))
+		// Stop the cockroach process gracefully in order to drain it properly.
+		// This makes the upgrade closer to how users do it in production, but
+		// it's also needed to eliminate flakiness. In particular, this will
+		// make sure that DistSQL draining information is dissipated through
+		// gossip so that other nodes running an older version don't consider
+		// this upgraded node for DistSQL plans (see #87154 for more details).
+		// TODO(yuzefovich): ideally, we would also check that the drain was
+		// successful since if it wasn't, then we might see flakes too.
```

Comment:
Currently, `StopCockroachGracefullyOnNode` will wait for 1 minute, and then SIGKILL the process. If we instead waited for one minute and bubbled up a timeout error if the process is still running, would that deal with the situation described here?